openupgradelib/openupgrade.py

```diff
@@ -166,11 +168,12 @@ def do_raise(error):
     'delete_record_translations',
     'disable_invalid_filters',
     'delete_records_safely_by_xml_id',
+    'search_and_replace_single_id',
 ]
 
 
 @contextmanager
-def allow_pgcodes(cr, *codes):
+def allow_pgcodes(cr, *codes, **kwargs):
```

Comment:
Python 3 will support a signature like (cr, *codes, query=None), but Python 2 does not.